File: KthLargestElement.java
import java.util.*;
// Approach 1- Priority Min-Heap
class KthLargest { // TC O(n log n) SC O(k)
    private static int k;
    private PriorityQueue<Integer> heap;
    
    public KthLargest(int k, int[] nums) {
        this.k = k;
        heap = new PriorityQueue<>();
        
        for (int num: nums) {
            heap.offer(num);
        }
        
        while (heap.size() > k) {
            heap.poll();
        }
    }
    
    public int add(int val) {
        heap.offer(val);
        if (heap.size() > k) {
            heap.poll();
        }

        return heap.peek();
    }
}
class KthLargestElement{
    public static void main(String[] args) {
        int[] nums={4,5,8,2};
        int k=3;
        KthLargest kth=new KthLargest(k, nums);
        System.out.println(kth.add(3));
        System.out.println(kth.add(5));
        System.out.println(kth.add(10));
        System.out.println(kth.add(9));
        System.out.println(kth.add(4));

    }
}

File: KthLargestElement.py
import heapq
class KthLargest:
# Approach 1- Priority Min-Heap
    def __init__(self, k, nums): # TC O(n log n) // SC O(k)
        self.k=k
        self.heap=nums
        
        heapq.heapify(self.heap)
        while len(self.heap)>self.k:
            heapq.heappop(self.heap)
            
    def add(self, val: int) -> int:
        heapq.heappush(self.heap, val)
        
        if len(self.heap)>self.k:
            heapq.heappop(self.heap)
        
        return self.heap[0]


# Your KthLargest object will be instantiated and called as such:
k=3
nums=[4,5,8,2]
kth = KthLargest(k, nums)
print(kth.add(3))
print(kth.add(5))
print(kth.add(10))
print(kth.add(9))
print(kth.add(4))